### PR TITLE
System Console settings restricted for Cloud

### PR DESCRIPTION
**Pull request by marianunez**
#### Summary

1. Hide the settings below in the System Console if `RestrictSystemAdmin` is enabled for Mattermost Cloud: 

- Server Logs
- Site Customization: 
	- Terms of Service link
	- Privacy Policy link
	- About link
	- Report a Problem link
	- Mattermost Apps Download Page Link
	- Android App Download Link
	- iOS App Download Link

- Users and Team:
	- Allow Team Administrators to edit others posts

- Plugin Management:
	- Enable Plugins
	- Require Plugin Signature
	- Enable Automatic Prepackaged Plugins
	- Upload Plugin (choose file)
	- Enable Marketplace
	- Enable Remote Marketplace
	- Marketplace URL

2. Removed restriction on the Open Server setting so Admins in Cloud can modify though System Console.

#### Related PRs

Server: https://github.com/mattermost/mattermost-server/pull/15528

#### Ticket Link
[MM-28834](https://mattermost.atlassian.net/browse/MM-28834)

### CODE REVIEW

File: components/admin_console/plugin_management/plugin_management.tsx [hunk 1]
```diff
@@ -1018,143 +1018,146 @@ export default class PluginManagement extends AdminSettings<Props, State> {
...
-                                <FormattedMessage
-                                    id='admin.plugins.settings.automaticPrepackagedPlugins'
-                                    defaultMessage='Enable Automatic Prepackaged Plugins:'
-                                />
-                            }
-                            helpText={
-                                <FormattedMarkdownMessage
-                                    id='admin.plugins.settings.automaticPrepackagedPluginsDesc'
-                                    defaultMessage='When true, automatically installs any prepackaged plugin found to be enabled in the server configuration.'
+                        { !this.props.config.ExperimentalSettings.RestrictSystemAdmin && (
```
Comment (marianunez) [inline components/admin_console/plugin_management/plugin_management.tsx:1021]: This was the only line added to this file. Github is not great handling all the indentation differences.

File: components/admin_console/plugin_management/plugin_management.tsx [hunk 2]
```diff
@@ -1018,143 +1018,146 @@ export default class PluginManagement extends AdminSettings<Props, State> {
...
-                                <FormattedMessage
-                                    id='admin.plugins.settings.automaticPrepackagedPlugins'
-                                    defaultMessage='Enable Automatic Prepackaged Plugins:'
-                                />
-                            }
-                            helpText={
-                                <FormattedMarkdownMessage
-                                    id='admin.plugins.settings.automaticPrepackagedPluginsDesc'
-                                    defaultMessage='When true, automatically installs any prepackaged plugin found to be enabled in the server configuration.'
+                        { !this.props.config.ExperimentalSettings?.RestrictSystemAdmin && (
```
Comment (marianunez) [inline components/admin_console/plugin_management/plugin_management.tsx:1021]: This was the only line added to this file. Github is not great handling all the indentation differences.

File: components/admin_console/admin_definition.jsx [hunk 3]
```diff
@@ -2438,7 +2451,6 @@ const AdminDefinition = {
                         label_default: 'Enable Open Server: ',
                         help_text: t('admin.team.openServerDescription'),
                         help_text_default: 'When true, anyone can signup for a user account on this server without the need to be invited.',
-                        isHidden: it.configIsTrue('ExperimentalSettings', 'RestrictSystemAdmin'),
```
Comment (marianunez) [inline components/admin_console/admin_definition.jsx:2441]: @jwilander Heads up this was restricted from before and I have removed it so they can now edit change if they want to have it as an Open Server.

**Comment by jwilander** [comment]
@marianunez we should also make sure that these settings cannot be updated via the API too

**Comment by marianunez** [comment]
> @marianunez we should also make sure that these settings cannot be updated via the API too

Yes, separate PR in the works.

File: components/admin_console/plugin_management/plugin_management.tsx (hunk 2 above)
Comment (ali-farooq0) [reply components/admin_console/plugin_management/plugin_management.tsx:1021]: Thanks Maria! helps to lower the number of lines i have to scan :D

File: components/admin_console/plugin_management/plugin_management.tsx [hunk 4]
```diff
@@ -1018,143 +1018,146 @@ export default class PluginManagement extends AdminSettings<Props, State> {
...
+                                    setByEnv={this.isSetByEnv('PluginSettings.MarketplaceUrl')}
                                 />
-                            }
-                            helpText={this.getMarketplaceUrlHelpText(this.state.marketplaceUrl)}
-                            value={this.state.marketplaceUrl}
-                            disabled={this.props.isDisabled || !this.state.enable || !this.state.enableMarketplace || !this.state.enableRemoteMarketplace}
-                            onChange={this.handleChange}
-                            setByEnv={this.isSetByEnv('PluginSettings.MarketplaceUrl')}
-                        />
+                            </>)}
```
Comment (ali-farooq0) [inline components/admin_console/plugin_management/plugin_management.tsx:1160]: minor nit: `}` should go on the next line

**Review by ali-farooq0 — APPROVED** [review]

**Comment by ali-farooq0** [comment]
ohh here comes the CI snapshot vultures :P

**Review by jwilander — APPROVED** [review]

**Comment by marianunez** [comment]
@stevemudie the test server is updated with the config setting `RestrictSystemAdmin` set to true ready for testing. 

Thanks @ali-farooq0 for updating it!

**Review by stevemudie — APPROVED** [review]
LGTM

**Pull request merged**